jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostConfigureSuite.scala

```diff
@@ -77,4 +77,17 @@ class XGBoostConfigureSuite extends FunSuite with PerTest {
     }
     ss.conf.unset("xgboost.spark.ignoreSsl")
   }
+
+  test("Check setting rabit configs no affect on prediction") {
```

Comment:
this test is not enough, we should test all passed in rabit params are taking effect in rabit layer

Reply:
I see, then we need new api in rabit and jni to read parameter out. like https://github.com/dmlc/rabit/commit/edac14220669bab9c3605ee1664d3bce816dbf3c

Reply:
or we can mock rabit.init

Reply:
giving a quick check neither of those options seems easy to add on this pr. I would prefer adding a follow up pr and adding unittest based on mockito to xgboost4j related projects. What do you think @CodingCat

Reply:
(1) the test here is not reasonable, why testing `no affect on prediction` is enough for this?

(2) why writing test is difficult?

(2) even writing test is difficult for whatever reason we should ensure we can write test first (i.e. finish whatever needed to adding test for this PR first), and then add this PR later

Reply:
I agree with (3), my concern is really if this diff gets too big to review.
(2) try to support powermock in spark tests, let met do more research on this

Reply:
added test verify Rabit get init with expected params

Reply:
moved all rabit tests to a separate suite